client/galaxy/scripts/components/History/HistoryDropdown.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+    <div>
+        <b-link
+            id="history-dropdown"
+            class="history-dropdown font-weight-bold"
+            data-toggle="dropdown"
+            aria-haspopup="true"
+            aria-expanded="false"
+        >
+            {{ history.name }}
+        </b-link>
+        <div class="dropdown-menu" aria-labelledby="history-dropdown">
+            <a class="dropdown-item" :href="urlView">View</a>
+            <a class="dropdown-item" href="#" @click="onSwitch">Switch To</a>
+            <a class="dropdown-item" :href="urlShowStructure">Structure</a>
+            <a class="dropdown-item" :href="urlSharing">Sharing</a>
+        </div>
+    </div>
+</template>
+<script>
+import { getAppRoot } from "onload/loadConfig";
+import { setCurrent } from "./services";
+
+export default {
+    props: ["history"],
+    data() {
+        return {
+            urlView: `${getAppRoot()}histories/view?id=${this.history.id}`,
+            urlShowStructure: `${getAppRoot()}histories/show_structure?id=${this.history.id}`,
+            urlSharing: `${getAppRoot()}histories/sharing?id=${this.history.id}`
+        };
+    },
+    created() {
+        this.root = getAppRoot();
+    },
+    methods: {
+        onSwitch: function() {
+            // this switches the history, but doesn't trigger a reload yet...
```

Comment:
I'm not sure if this is correct and at which level to place this, but `Galaxy.currHistoryPanel.switchToHistory(this.history.id);` seems to work in my testing.

Reply:
Yeah - I think I'm going to revert the service extraction and de-duplication and just use this line. I'm worried about conflicts with other history panel work on going - this seems to be a very clean solution - thanks!